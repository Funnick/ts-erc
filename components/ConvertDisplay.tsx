import { Change, ConvertText, FlexContainer,
    Column, CenterText, IconColumn, Text } from './styled/styledComponents';

type ConvertDisplayProps = {
    prevAmount: number
    prevFromCurrency: string
    converted?: number
    prevToCurrency: string
    converting: boolean
}

const ConvertDisplay = ({ prevAmount, prevFromCurrency, converted, prevToCurrency, converting }: ConvertDisplayProps) => {
return (
    <Change>
        { converting ? 
        <ConvertText>Converting...</ConvertText> :
        <FlexContainer>
            <Column>
                <CenterText>
                    {prevAmount} {prevFromCurrency}
                </CenterText>
            </Column>
            <IconColumn>
                <Text>
                =
                </Text>
            </IconColumn>
            <Column>
                <CenterText>
                    {converted} {prevToCurrency}
                </CenterText>
            </Column>
        </FlexContainer>
        }
    </Change>
)}

export default ConvertDisplay