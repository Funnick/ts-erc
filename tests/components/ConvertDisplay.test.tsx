import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ConvertDisplay from '@/components/ConvertDisplay';

describe('Component Test ConvertDisplay', () => {
    it('Should render a "Converting..." text', () => {
        render(<ConvertDisplay prevAmount={1}
        prevFromCurrency={"EUR"} converted={5}
        prevToCurrency={"USD"} converting={true} />)
        
        const text = screen.getByText("Converting...")
        
        expect(text).toBeInTheDocument()
    })

    it('Should render a change', () => {
        render(<ConvertDisplay prevAmount={1}
        prevFromCurrency={"EUR"} converted={5}
        prevToCurrency={"USD"} converting={false} />)

        const from = screen.getByText('1 EUR')
        const to = screen.getByText('5 USD')
        
        expect(from).toBeInTheDocument()
        expect(to).toBeInTheDocument()
    })

    it('Snapshot_1', () => {
        const tree = renderer.create(
        <ConvertDisplay prevAmount={1}
        prevFromCurrency={"EUR"} converted={5}
        prevToCurrency={"USD"} converting={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Snapshot_2', () => {
        const tree = renderer.create(
        <ConvertDisplay prevAmount={1}
        prevFromCurrency={"EUR"} converted={5}
        prevToCurrency={"USD"} converting={false} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
}) 