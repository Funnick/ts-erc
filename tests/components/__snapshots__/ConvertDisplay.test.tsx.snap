// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test ConvertDisplay Snapshot_1 1`] = `
<div
  className="sc-jJTsDX hVWvFj"
>
  <p
    className="sc-fFGjHI sc-dicizt sc-iVCKna fxeOjw chuGhL bwBmdQ"
  >
    Converting...
  </p>
</div>
`;

exports[`Component Test ConvertDisplay Snapshot_2 1`] = `
<div
  className="sc-jJTsDX hVWvFj"
>
  <div
    className="sc-kPTPQs dVARFL"
  >
    <div
      className="sc-ispOId duzTmJ"
    >
      <p
        className="sc-fFGjHI sc-dicizt fxeOjw chuGhL"
      >
        1
         
        EUR
      </p>
    </div>
    <div
      className="sc-ispOId sc-jrrXlR duzTmJ gjvJFm"
    >
      <p
        className="sc-fFGjHI fxeOjw"
      >
        =
      </p>
    </div>
    <div
      className="sc-ispOId duzTmJ"
    >
      <p
        className="sc-fFGjHI sc-dicizt fxeOjw chuGhL"
      >
        5
         
        USD
      </p>
    </div>
  </div>
</div>
`;
